lib/pbench/server/api/__init__.py

```diff
@@ -323,21 +332,30 @@ def post(self):
                     f.write(chunk)
                     hash_md5.update(chunk)
         except Exception:
-            app.logger.exception("There was something wrong uploading {}", filename)
+            app.logger.exception(
+                "Tarfile upload: There was something wrong uploading {}", filename
+            )
             abort(500, message=f"There was something wrong uploading {filename}")
 
         message = None
-        if bytes_received > content_length:
-            message = f"More bytes received than the specified content length {content_length}"
+        if bytes_received != content_length:
+            app.logger.debug(
+                "Tarfile upload: Bytes received does not match with content length header value"
+            )
+            message = (
+                f"Bytes received ({bytes_received}) does not match with content length header"
+                f" ({content_length}), upload failed"
+            )
 
         if hash_md5.hexdigest() != md5sum:
-            message = f"md5sum check failed for {filename}"
+            app.logger.debug("Tarfile upload: md5sum check failed")
+            message = f"md5sum check failed for {filename}, upload failed"
 
         if message:
             try:
                 os.remove(full_path)
             except Exception as exc:
-                self.logger.warning(
+                app.logger.debug(
```

Comment:
Is there a reason to change this to a `debug` message?  Seems like any failure to clean up should be flagged as a warning as it could indicate other problems.